### ISSUE TITLE
cmake install support request

### ISSUE THREAD

**Issue by justinpa-asg** [hazelcast/hazelcast-cpp-client#374]
Please append to CMakeLists.txt:
```
INSTALL(DIRECTORY hazelcast/include/ DESTINATION include)
INSTALL(DIRECTORY hazelcast/generated-sources/include/ DESTINATION include)
INSTALL(DIRECTORY external/include/boost/ DESTINATION include/boost)
INSTALL(TARGETS ${HZ_LIB_NAME} LIBRARY DESTINATION lib)
```

cmake will link with -rpath unless you install.

**Comment by ihsandemir** [comment]
will be fixed by #723

**Comment by ihsandemir** [comment]
fixed by #723
